packages/cozy-doctypes/src/contacts/Contact.js

```diff
@@ -0,0 +1,37 @@
+const getPrimaryOrFirst = property => obj => {
+  if (!obj[property] || obj[property].length === 0) return ''
+  return obj[property].find(property => property.primary) || obj[property][0]
+}
+
+class Contact extends Document {
+  static getInitials(name) {
+    if (!name) return ''
+    if (typeof name === 'string') {
+      return name[0].toUpperCase()
+    }
+
+    return ['givenName', 'familyName']
+      .map(part => name[part] || '')
+      .map(name => name[0])
+      .join('')
+      .toUpperCase()
+  }
+
+  // TODO: sadly we have different versions of contacts' doctype to handle...
+  // A migration tool on the stack side is needed here
+  static getPrimaryEmail(contact) {
+    return Array.isArray(contact.email)
+      ? getPrimaryOrFirst('email')(contact).address
+      : contact.email
+  }
+
+  static getPrimaryCozy(contact) {
+    return Array.isArray(contact.cozy)
+      ? getPrimaryOrFirst('cozy')(contact).url
+      : contact.url
+  }
+}
```

Comment:
we should also add getDisplayName ?

Reply:
I thought about it but I think that `getDisplayName` has more to do with recipients than with contacts (sometimes recipient is a contact, sometimes it is the cozy "owner" and maybe sometimes something else, I take a look)